tests/foreman/cli/test_repository_set.py

```diff
@@ -147,3 +149,142 @@ def test_repositoryset_available_repositories(self):
             sum(int(repo['enabled'] == u'true') for repo in result.stdout),
             0
         )
+
+    def test_repositoryset_disable_by_name(self):
+        """@Test: Disable repo from reposet by names of reposet, org and product
+
+        @Feature: Repository-set
+
+        @Assert: Repository was disabled
+
+        """
+        org = make_org()
+        manifest = manifests.clone()
+        upload_file(manifest, remote_file=manifest)
+        result = Subscription.upload({
+            u'file': manifest,
+            u'organization-id': org['id'],
+        })
+        self.assertEqual(result.return_code, 0)
+        result = RepositorySet.enable({
+            u'name': REPOSET['rhva6'],
+            u'organization': org['name'],
+            u'product': PRDS['rhel'],
+            u'releasever': '6Server',
+            u'basearch': 'x86_64',
+        })
+        self.assertEqual(result.return_code, 0)
+        result = RepositorySet.disable({
+            u'name': REPOSET['rhva6'],
+            u'organization': org['name'],
+            u'product': PRDS['rhel'],
+            u'releasever': '6Server',
+            u'basearch': 'x86_64',
+        })
+        self.assertEqual(result.return_code, 0)
+        result = RepositorySet.available_repositories({
+            u'name': REPOSET['rhva6'],
+            u'organization': org['name'],
+            u'product': PRDS['rhel'],
+        })
+        self.assertEqual(result.return_code, 0)
+        for repo in result.stdout:
+            if repo['release'] == '6Server' and repo['arch'] == 'x86_64':
+                enabled = repo['enabled']
+        self.assertEqual(enabled, 'false')
```

Comment:
As in #2457, the `enabled` variable will be unset if the conditional above is not satisfied.